languages/python/exercises/concept/strings/.docs/hints.md

```diff
@@ -0,0 +1,22 @@
+# Hints
+
+### General
+
+- The [Python documentation for `str`][https://docs.python.org/3.7/library/stdtypes.html#text-sequence-type-str] is an overview of the Python `str` type.
+
+### 1. Get message from a log line
+
+ - Strings in Python have [lots of convenient instance methods][https://docs.python.org/3/library/stdtypes.html#str] for cleaning, splitting, manipulating, and creating new strings. Extracting values from a string could be done by splitting it based on a substring, for example.
+
+### 2. Get log level from a log line
+
+ - Strings also have methods that help convert letters from lower to uppercase and vice-versa.
+
+### 3. Reformat a log line
+
+Strings are immutable, but can be combined together to make new strings, or have elements replaced. This goal can be accomplished by using string methods, or operators like `+` or `+=` (which are overloaded to work with strings).
+Python also has a concept of string formatting, like many other languages.
+
+ - The [`str.join()`][https://docs.python.org/3/library/stdtypes.html#str.join] method is useful to concatenate an iterable of strings into one string.
```

Comment:
It might be good to mention that `str.join` is more specifically about intercalating / interspersing multiple strings around a common delimiter ... it's not _exactly_ concatenation, although joining around the empty string does give you the same result.